android/build.gradle

```diff
@@ -11,6 +11,15 @@ ext {
     targetSdkVersion    = Integer.valueOf(project.targetSdkVersion)
     supportLibVersion   = project.supportLibVersion
     gradlePluginVersion = project.gradlePluginVersion
+
+    if (System.properties['os.arch'] == "aarch64") {
+        // For M1 Users we need to use the NDK 24 which added support for aarch64
+        ndkVersion = "24.0.8215888"
+    } else {
+        // Otherwise we default to the side-by-side NDK version from AGP.
```

Comment:
What happens if we use NDK 24 for non `aarch64` on Darwin?

Reply:
This fix was recommended in the rn-upgrade helper to reach 0.68.5

For the react-native version in this PR -> 0.67.5, NDK 24 works well without any issues.
I recently tried it out here : https://github.com/status-im/status-mobile/pull/15483/commits/c6ca6a0d8565edfb9c6e5b407a268277f862f141

However I remember getting NDK version related build errors back when I had upgraded to react-native 0.68.5